src/router.ts

```diff
@@ -206,13 +211,25 @@ const introduceNewVoterToSlackChannel = async (
   if (process.env.CLIENT_ORGANIZATION === 'VOTE_AMERICA') {
     if (includeWelcome) {
       // Voter initiated conversation with "HELPLINE".
-      messageToVoter = MessageConstants.WELCOME_AND_STATE_QUESTION();
+      if (userInfo.stateName) {
+        messageToVoter = MessageConstants.WELCOME_FINDING_VOLUNTEER();
+      } else {
+        messageToVoter = MessageConstants.WELCOME_AND_STATE_QUESTION();
+      }
     } else {
       // Voter has already received automated welcome and is just now responding with "HELPLINE".
-      messageToVoter = MessageConstants.STATE_QUESTION();
+      if (userInfo.stateName) {
+        messageToVoter = MessageConstants.NO_STATE_FINDING_VOLUNTEER();
+      } else {
+        messageToVoter = MessageConstants.STATE_QUESTION();
+      }
     }
   } else {
-    messageToVoter = MessageConstants.WELCOME_VOTER();
+    if (userInfo.stateName) {
```

Comment:
This else branch is for non-VOTE_AMERICA right? Would userInfo.stateName ever be already populated at this point for other clients?

Reply:
I see at least one case where it might be? https://github.com/tomerovadia/VoterHelpLine/blob/master/src/router.ts#L404  I don't really understand what the PUSH/PULL stuff is all about or how other orgs are using it.

Reply:
Ya sorry I know the PUSH/PULL is confusing. When handleNewVoter is called with a PUSH entrypoint, the state is determined but no message is sent to the voter (below, the `TwilioApiUtil.sendMessage` is guarded by `entryPoint === LoadBalancer.PULL_ENTRY_POINT`.

So basically, the state will only be determined in PUSH entrypoint cases, and a message to voter will only happen in PULL entrypoint cases. So you can leave this as only `messageToVoter = MessageConstants.WELCOME_VOTER();` because messageToVoter will only be used `userInfo.stateName` is null anyway.

Hope that makes sense, but feel free to ask me to clarify more if needed!

Reply:
Makes sense!